balancer/balancer.go

```diff
@@ -380,7 +381,8 @@ type ConnectivityStateEvaluator struct {
 //  - If at least one SubConn in Ready, the aggregated state is Ready;
 //  - Else if at least one SubConn in Connecting, the aggregated state is Connecting;
 //  - Else if at least one SubConn is TransientFailure, the aggregated state is Transient Failure;
-//  - Else the aggregated state is Idle
+//  - Else if at least one SubConn is Idle, the aggregated state is Idle;
```

Comment:
If there's one SubConn, and it's failing, the overall state will keep switching back to Idle?

This is not a big problem for policies that use this (e.g. roundrobin).
We may still want to distinguish between the initial Idle and the Idle after transient failure. But that needs to keep track of SubConns and their states.

Reply:
RR-type balancers should probably all implement the "once TF, always TF until ready" semantic.

I'm not sure what change is being requested here.

Reply:
> RR-type balancers should probably all implement the "once TF, always TF until ready" semantic.

Oof, this made me realize that our implementation of that logic is now broken with idleness, as it was not implemented via "newState != Ready/Shutdown", but "newState == Connecting".  I've updated the logic to handle that now.